src/Std/Data/DTreeMap/Lemmas.lean

```diff
@@ -661,6 +678,19 @@ theorem getKey?_eq_some_getKey [TransCmp cmp] {a : α} {h'} :
     t.getKey? a = some (t.getKey a h') :=
   Impl.getKey?_eq_some_getKey t.wf
 
+theorem compare_getKey_self [TransCmp cmp] {k : α} (h' : k ∈ t) :
+    cmp (t.getKey k h') k = .eq :=
+  Impl.compare_getKey_self t.wf h'
+
+theorem getKey_congr [TransCmp cmp] {k₁ k₂ : α} (h' : cmp k₁ k₂ = .eq)
+    (h₁ : k₁ ∈ t) : t.getKey k₁ h₁ = t.getKey k₂ ((mem_congr h').mp h₁) :=
+  Impl.getKey_congr t.wf h' h₁
+
+@[simp]
+theorem getKey_eq [TransCmp cmp] [LawfulEqCmp cmp] [BEq α] [LawfulBEqCmp cmp] {k : α}
```

Comment:
Where does the `BEq` instance come into play here? Same question about many other lemmas in this file.